doc/specifications/proposed/content_download/content_download.md

```diff
@@ -0,0 +1,44 @@
+# Content download
+
+> Author: bd@ez.no
+
+## Summary
+This feature covers downloading of Content binary files over HTTP.
+
+It covers:
+- generation of proper links from Twig as well as PHP,
+- permissions checking (the user must be allowed to read the Content)
+- streaming of the file over HTTP
+
+## Components
+
+### Twig helper
+
+> Name: ez_download_path
+> Usage: `ez_download_path( content, 'file' )`
+
+The helper will be used to generate a download link for a given Field/Content.
+
+### Controller action + route
+
+> Route: `/content/download/{fileId}/{fileName}`
```

Comment:
@crevillo you probably should add per-language as well :-) (or is the choice of language dictated by http headers in the request? not sure I'd go that route)

If the fileId is generated by the system by some opaque hashing, this means that
- we can save links to archived versions (good)
- we can not have a fixed link to the latest version in language X (very bad)

If the fileId is otho an identifier of the content, we miss in the url the selectors for field/language/version

If fileid is ad identifier for the content field, we still miss selectors for language/version